### PR TITLE
UCP: Migrate `GreatestInt` (Vectorize) from TiDB

### LINKED ISSUE #6743: UCP: Migrate `GreatestInt` (Vectorize) from TiDB
## Description

Port the function `GreatestInt`  (Vectorize)  from TiDB to coprocessor.

## Score 

* 50

## Mentor(s)

* @breeswish 

## Recommended Skills

* Rust programming 

## Learning Materials

Functions ported from TiDB

- https://github.com/tikv/tikv/tree/master/components/tidb_query/src/rpn_expr) 
- https://github.com/tikv/tikv/tree/master/components/tidb_query/src/expr)

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

UCP: #6743 

### What problem does this PR solve?

Issue Number: close #6743 

Problem Summary:

Port the function `GreatestInt` (Vectorize) from TiDB to coprocessor.

### What is changed and how it works?

* Add vectorized GreatestInt
* Add vectorized GreatestInt unit tests

### Check List 

Tests

- Unit test

### CODE REVIEW

File: components/tidb_query_vec_expr/src/impl_compare.rs [hunk 1]
```diff
@@ -227,6 +227,38 @@ pub fn coalesce<T: Evaluable>(args: &[&Option<T>]) -> Result<Option<T>> {
     Ok(None)
 }
 
+#[rpn_fn(varg)]
```
Comment (sticnarf) [inline components/tidb_query_vec_expr/src/impl_compare.rs:230]: I think it's vital to validate if the argument count is at least 2

Comment (JmPotato) [reply components/tidb_query_vec_expr/src/impl_compare.rs:230]: You're right about the arguments count validation, but why at least 2? I think one argument should be ok for this func.

BTW, to validate the argument count, is `#[rpn_fn(varg, min_args = 1)]` right to do this?

Comment (sticnarf) [reply components/tidb_query_vec_expr/src/impl_compare.rs:230]: Because `GREATEST()` in MySQL just requires there are at least two arguments...

Comment (sticnarf) [reply components/tidb_query_vec_expr/src/impl_compare.rs:230]: Yes. The way you validate it is right.

Comment (JmPotato) [reply components/tidb_query_vec_expr/src/impl_compare.rs]: Thanks for point out my mistake. In this case, should I remove the single argument unit test cases also?

Comment (sticnarf) [reply components/tidb_query_vec_expr/src/impl_compare.rs]: Ah, yes. You'd better remove it as it's an invalid case.

**Comment by JmPotato** [comment]
Seems like the test platform has an internal problem?

```shell
java.nio.channels.ClosedChannelException

```

**Comment by zhongzc** [comment]
Congrats!

**Pull request merged**